### PR TITLE
feat: Implement Password based Login

### PR DESCRIPTION
**Pull request by nam-t24**
Features Added
- login and signup page
- password based login/signup option
- email verification (requirement before logging in)
- password reset email to user
- browser save account credential and autocomplete
- challenge card masonry layout in hackcenter

### CODE REVIEW

**Comment by abdullah50053** [comment]
Screen Shot 2022-02-02 at 9 30 53 AM
Password should not be visible on account creation.

**Comment by nam-t24** [comment]
> Screen Shot 2022-02-02 at 9 30 53 AM
> Password should not be visible on account creation.

When they create an account shouldn't they know what password they're entering in incase they mistype a character?

**Comment by nam-t24** [comment]
I could have the password field have a checkbox to allow the user to mask or unmask the password they are typing. Eliminates the confirm password field for better user experience.

File: pages/auth/index.tsx [hunk 1]
```diff
@@ -8,32 +11,110 @@ import { useAuthContext } from '../../lib/user/AuthContext';
...
+          <div className="text-sm">or</div>
+          <div className="w-[24rem]">
+            <input
+              className="w-full rounded-lg p-2 border-2 border-gray-500"
+              value={currentEmail}
+              onChange={(e) => setCurrentEmail(e.target.value)}
+              style={{ backgroundColor: '#FFF' }}
+              placeholder="Email"
+            ></input>
+            <input
```
Comment (LoganLieou) [inline pages/auth/index.tsx]: type="password"

**Review by LoganLieou — CHANGES_REQUESTED** [review]

**Comment by WillieCubed** [comment]
#### 📝 Changed routes:


| [**/admin**](https://hackportal-svdd5ogvf-acmutd.vercel.app/admin) | [**…n/resolve/\[questionId\]**](https://hackportal-svdd5ogvf-acmutd.vercel.app/admin/resolve/[questionId]) |
|:-:|:-:|
|<a href="https://hackportal-svdd5ogvf-acmutd.vercel.app/admin">Screenshot of /admin</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/tyKo1fSY1vp-4rIPaCvZ9zE9Kx29KPs-3H8XiQSKS66Mex74IjxYhftjo9zm-K_RCZjNoKiJOyc8FMnDp1vBGg.png">(view full size)</a> |<a href="https://hackportal-svdd5ogvf-acmutd.vercel.app/admin/resolve/[questionId]">Screenshot of /admin/resolve/[questionId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/h38-cI22HJzg-u9XZNdynpTFhKvtTyAUlC_ED8vQcYkxzGEehVBhte6TJ-sWLEENc-vVSASlGivMG-bs7TfgrA.png">(view full size)</a> |



| [**/admin/scan**](https://hackportal-svdd5ogvf-acmutd.vercel.app/admin/scan) | [**/admin/users**](https://hackportal-svdd5ogvf-acmutd.vercel.app/admin/users) |
|:-:|:-:|
|<a href="https://hackportal-svdd5ogvf-acmutd.vercel.app/admin/scan">Screenshot of /admin/scan</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/y1hVSF44ldex0wC4t-vRibp_AK25yp_EMwSxE0y51Toy5JpYORpm3vyKX43chWGxtKaD0DdDnuvN6ivm64sgsA.png">(view full size)</a> |<a href="https://hackportal-svdd5ogvf-acmutd.vercel.app/admin/users">Screenshot of /admin/users</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/nd2IWK-HY2UHtIgQhOXxyezbbCB0cmThcjcd5YC5w7H3ms21NX_FYkb5vkvK6OstfQ6dJo71GTfaXDMZ_O2_NA.png">(view full size)</a> |



| [**/auth**](https://hackportal-svdd5ogvf-acmutd.vercel.app/auth) | [**/auth/signup**](https://hackportal-svdd5ogvf-acmutd.vercel.app/auth/signup) |
|:-:|:-:|
|<a href="https://hackportal-svdd5ogvf-acmutd.vercel.app/auth">Screenshot of /auth</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/d1J8CvCuC5QHBXLTpoPfAT_uq0QSd1PDKiO5V21pc9DXYXJtEqSt0R4v4EMxALICT4wulmUBvQfVhzY84wJh1g.png">(view full size)</a> |<a href="https://hackportal-svdd5ogvf-acmutd.vercel.app/auth/signup">Screenshot of /auth/signup</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/AV0kITg15B-MBGnhQ1wMtILf9_UkEVKBITlcFeNlLRMt6yNma1DOTJ6d_i9c4xAMPs1S7fXi7G0x52h2EgmHeQ.png">(view full size)</a> |



And 16 other routes:
- [**/api/applications/\[applicationId\]**](https://hackportal-svdd5ogvf-acmutd.vercel.app/api/applications/[applicationId])
- [**/api/applications**](https://hackportal-svdd5ogvf-acmutd.vercel.app/api/applications)
- [**/api/scan**](https://hackportal-svdd5ogvf-acmutd.vercel.app/api/scan)
- [**/api/scan/create**](https://hackportal-svdd5ogvf-acmutd.vercel.app/api/scan/create)
- [**/api/scan/delete**](https://hackportal-svdd5ogvf-acmutd.vercel.app/api/scan/delete)
- [**/api/scan/update**](https://hackportal-svdd5ogvf-acmutd.vercel.app/api/scan/update)
- [**/api/userinfo**](https://hackportal-svdd5ogvf-acmutd.vercel.app/api/userinfo)
- [**/api/users/roles**](https://hackportal-svdd5ogvf-acmutd.vercel.app/api/users/roles)
- [**/dashboard/Components/ChallengeCard**](https://hackportal-svdd5ogvf-acmutd.vercel.app/dashboard/Components/ChallengeCard)
- [**/dashboard/Components/Sidebar**](https://hackportal-svdd5ogvf-acmutd.vercel.app/dashboard/Components/Sidebar)
- [**/dashboard/hackerpack**](https://hackportal-svdd5ogvf-acmutd.vercel.app/dashboard/hackerpack)
- [**/dashboard**](https://hackportal-svdd5ogvf-acmutd.vercel.app/dashboard)
- [**/dashboard/scan-in**](https://hackportal-svdd5ogvf-acmutd.vercel.app/dashboard/scan-in)
- [**/**](https://hackportal-svdd5ogvf-acmutd.vercel.app/)
- [**/profile**](https://hackportal-svdd5ogvf-acmutd.vercel.app/profile)
- [**/register**](https://hackportal-svdd5ogvf-acmutd.vercel.app/register)



Commit 95312bd9a42ee8438e9ff76cf51cea0244b5cad0 (https://hackportal-svdd5ogvf-acmutd.vercel.app).

**Comment by abdullah50053** [comment]
Need merge conflicts resolved @nam-t24

**Review by abdullah50053 — APPROVED** [review]

**Review by DedsecKnight — APPROVED** [review]

**Pull request merged**